### PR TITLE
Align the behavior of DynamicRealm.executeTransaction() to Realm.executeTransaction()

### PR DESCRIPTION
**Pull request by zaki50**
please review this @realm/java

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/DynamicRealm.java [hunk 1]
```diff
@@ -130,20 +131,21 @@ public void clear(String className) {
      * during the transaction {@link #cancelTransaction()} will be called instead of {@link #commitTransaction()}.
      *
      * @param transaction {@link io.realm.DynamicRealm.Transaction} to execute.
-     * @throws RealmException if any error happened during the transaction.
      */
     public void executeTransaction(Transaction transaction) {
         if (transaction == null)
-            return;
+            throw new IllegalArgumentException("Transaction should not be null");
+
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/DynamicRealm.java]: Missing `}`

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,7 @@
+0.88.0
+ * BREAKING CHANGE: DynamicRealm.executeTransaction() now directly throws any RuntimeException instead of wrapping it in a RealmException (#1682).
+ * BREAKING CHANGE: DynamicRealm.executeTransaction() throws IllegalArgumentException if the Transaction object is null.
```
Comment (cmelchior) [inline changelog.txt]: now throws IllegalArgumentException instead of silently accepting a null Transaction object.

**Comment by cmelchior** [comment]
Good one. We forgot this when refactoring the standard `Realm` one. Apart from the small comments :+1:

**Comment by kneth** [comment]
:+1:

**Comment by cmelchior** [comment]
:+1:

**Pull request merged**